### PR TITLE
lang: Add warning sclang config file directory

### PR DESCRIPTION
**Pull request by redFrik**
the main reason for this is to warn users if a Quark didn't install correctly.
tested under osx and linux.

### CODE REVIEW

**Comment by brianlheim** [comment]
Can I ask, why just a warning? The whole point of this method is to create a file; if that fails I really think it should throw.

**Comment by redFrik** [comment]
i don't know how to throw an error in the proper way - if and what things need to be cleaned up etc.
also i'm not sure we agreed on that an error is the best solution.  thereby this middle ground harmless fix for now that i hope can go in quick.
(i wouldn't have bothered if this false positive install #2919 wasn't a real issue. but i do believe this is what caused me quite some woes during my teaching last semester.  a simple warning would've been very much appreciated.)

**Comment by brianlheim** [comment]
> also i'm not sure we agreed on that an error is the best solution. thereby this middle ground harmless fix for now that i hope can go in quick.

I think we should then discuss what the best solution is, there's no need to rush. I gave a few reasons already as to why I think it should be an error - it matches existing convention, and indicates an actual failure of the method. Perhaps in another design we would return a Boolean to say whether the operation succeeded, but that isn't the convention in the core lib, at least around file operations.

A warning is good for situations where the function is still able to complete, but maybe didn't do what the user wanted or expected.

You can look at the File primitives to see some examples of how to throw errors from sclang primitives. At minimum it's just a matter of returning an error code.

**Comment by redFrik** [comment]
you're not in a rush but for me this is urgent and important. my aim with this PR is to do something about the problem now and not losing it to endless discussions.  what's the harm?  it'll give us time to agree on the correct way.

**Comment by brianlheim** [comment]
I understand this is an important for you. However, it's only been a couple hours since you opened this, so I don't think this PR is at all in danger yet of becoming an endless discussion!

There are two ways I can respond to "what's the harm"? One is that we could get into the habit of adding technical debt with small and imperfect fixes, without ever going back to patch things up. @jamshark70 has brought this up a few times before, and I agree with him. The other is that a common complaint about SuperCollider is that its behavior is inconsistent; this change will add to that perception in my opinion.

Finally, I just don't think this issue is urgent enough to warrant merging without discussion. From what I've seen we reserve that urgency for severe-tagged issues only, which this definitely isn't. I think we have to be fair to everyone and make sure every change gets a period for concerns to be addressed.

**Comment by redFrik** [comment]
my idea with this deliberately (!) small and imperfect PR was that it'd be included in whatever next release will be.

but clearly i can't make myself understood. i did not mean that this PR is merge-asap-urgent, i meant that the #2919 issue is urgent and shouldn't be lost to endless discussion. besides this PR is absolutely harmless, doesn't feature creep, hastily bugfix or patch anything. it's a simple temporary warning.

but i get the hint - it's not going to happen.

File: lang/LangSource/SC_LanguageConfig.cpp [hunk 1]
```diff
@@ -189,6 +189,10 @@ bool SC_LanguageConfig::writeLibraryConfigYAML(const Path& fileName)
 	out << Value << gPostInlineWarnings;
 
 	out << EndMap;
+	
+	bfs::path dir(fileName.parent_path());
+	if(!(bfs::exists(dir)))
+		postfl("WARNING: Cannot create config file. First create the directory '%s'\n", dir.c_str());
```
Comment (telephon) [inline lang/LangSource/SC_LanguageConfig.cpp]: IIUC, you just add here:

`return errFailed;`

Comment (brianlheim) [reply lang/LangSource/SC_LanguageConfig.cpp]: the check needs to go in the primitive (this is a routine called during the store primitive) but yes

Comment (telephon) [reply lang/LangSource/SC_LanguageConfig.cpp]: maybe the easiest is to first merge this and then add that.

**Comment by brianlheim** [comment]
> my idea with this deliberately (!) small and imperfect PR was that it'd be included in whatever next release will be.

Absolutely, it still can be!

> but clearly i can't make myself understood. i did not mean that this PR is merge-asap-urgent, i meant that the #2919 issue is urgent and shouldn't be lost to endless discussion.

I'm sorry you feel that way. I've been trying to respond clearly to you but I'm sorry if I misunderstood. At the same time, I would like to be honest and say that I think you've been a bit stubborn in these conversations, and that's perhaps why they've dragged on. For example, in this PR I've given several reasons why I want a change, and you've never responded to that. How can we make progress that way?

> but i get the hint - it's not going to happen.

I didn't mean to give that impression, and I'm sorry if I did. I asked for a very small but in my view important change which is on the same order as what you've already done, and which I daresay would have taken much less time out of your day than this metadiscussion. You could still make those changes, or on the other hand, you could respond to the concerns I've raised. However, I agree that if you continue to ignore collaborators' input, this PR will never get merged.

**Comment by brianlheim** [comment]
Stale, superceded by #3577.

**Pull request closed without merging**